src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(OperatorIndexerCompletionProvider), LanguageNames.CSharp)]
+    [ExtensionOrder(After = nameof(SymbolCompletionProvider))]
+    [Shared]
+    internal class OperatorIndexerCompletionProvider : LSPCompletionProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public OperatorIndexerCompletionProvider()
+        {
+        }
+
+        private static readonly ImmutableHashSet<char> s_triggerCharactersHashSet = ImmutableHashSet.Create('.');
+
+        internal override ImmutableHashSet<char> TriggerCharacters => s_triggerCharactersHashSet;
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            if (token.IsKind(SyntaxKind.DotToken))
+            {
+                var parent = token.Parent;
+                var expression = parent switch
+                {
+                    MemberAccessExpressionSyntax memberAccess => memberAccess.Expression,
+                    _ => null,
+                };
+                if (expression != null)
+                {
+                    var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
+                    var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+                    var allMembers = container.GetMembers();
```

Comment:
This is in part a duplication of code that is done in `CSharpRecommendationServiceRunner`. We could either change the CSharpRecommendationServiceRunner to add the symbols if appropriate (see the changes in CSharpRecommendationServiceRunner later) or do the calculation here.